llm/llm.go

```diff
@@ -89,9 +89,10 @@ func New(workDir, model string, adapters []string, opts api.Options) (LLM, error
 		opts.NumGQA = 0
 		opts.RopeFrequencyBase = 0.0
 		opts.RopeFrequencyScale = 0.0
-		return newLlama(model, adapters, chooseRunners(workDir, "gguf"), ggml.NumLayers(), opts)
+		// return newLlama(model, adapters, chooseRunners(workDir, "gguf"), ggml.NumLayers(), opts)
+		return newLlamaExtServer(model, adapters, ggml.NumLayers(), opts)
 	case "ggml", "ggmf", "ggjt", "ggla":
-		return newLlama(model, adapters, chooseRunners(workDir, "ggml"), ggml.NumLayers(), opts)
+		return nil, fmt.Errorf("ggml model format no longer supported.  Please upgrade your model <URL here>")
```

Comment:
Keep an eye on https://github.com/jmorganca/ollama/pull/1268 for the experience deprecating the ggml runner.

cc @BruceMacD – not sure if you guys want merge these two branches or keep them separate? either works

Reply:
I think we should merge them separately otherwise this will be the mother of all changes, but I also don't want to railroad this change. I'll co-ordinate with Daniel.

Reply:
Let's get @BruceMacD's change in first, and I'll rebase on top of that, and pick up the MainGPU opt as well.